### DISCUSSION TITLE
Ip based case

### DISCUSSION THREAD

**Discussion by HrnSvnc** [vladimir-bukhtoyarov/bucket4j#172] [Q&A]
Hi everyone,
currently im trying to implement a limit based bucket by IPs. My idea was to throttle my api like this:

Every IP can request 25 Tokens per Minute but with a maximum of 2 token per second. 

My bucket configuration looks likes this:

```java
this.bucket = Bucket4j.builder()
                .addLimit(Bandwidth.classic(25, Refill.intervally(25,Duration.ofMinutes(1))))
                .addLimit(Bandwidth.simple(2,Duration.ofSeconds(1)))
                .build();
```
My method inside my controller looks like this:
```java
if(bucket.tryConsume(1)){
            return //
        }else{
            return ResponseEntity
                    .status(HttpStatus.TOO_MANY_REQUESTS)
                    .build();
        }
```
I have trouble implementing the ip based rule since Im not sure how that could work or if that functionality even exists. [The spring-boot-starter from Marc Griffing ](https://github.com/MarcGiffing/bucket4j-spring-boot-starter) does have the possibility to [use a filter Type based on IP adresses](https://github.com/MarcGiffing/bucket4j-spring-boot-starter#bucket4j_complete_properties).
Is there a way with bucket4j-core to use the same functionality?

Thanks in advance and thank you for your work!

**Comment by vladimir-bukhtoyarov** [comment]
Hello @HrnSvnc 

There is no such high-level functionality. What concrete troubles do you have?

**Reply by HrnSvnc** [reply]
Hi @vladimir-bukhtoyarov ,

My concrete problem is, that I want to throttle the usage of my api based on ip addresses. My api should be free to use for a period of time. After that I will use api keys. 

Would you have any suggestion on free and open apis throttling for not registered user and without an api key?

Thanks for the answer.
